Source/Csla.AspNetCore/Blazor/ApplicationContextManagerBlazor.cs

```diff
@@ -137,19 +141,33 @@ public IPrincipal GetUser()
     /// Not supported in Blazor.
     /// </summary>
     /// <param name="principal">Principal object.</param>
-    public virtual void SetUser(IPrincipal principal) => throw new NotSupportedException(nameof(SetUser));
+    public virtual void SetUser(IPrincipal principal)
+    {
+      ArgumentNullException.ThrowIfNull(principal);
```

Comment:
I suggest to add the following after the `null` check:
```csharp
if(principal is not ClaimsPrincipal claimsPrincipal)
{
  throw new NotSupportedException("The user to be set must be of type ClaimsPrincipal");
}
```
That would make a more actionable exception than an invalid cast in line 149 :)

Reply:
I did as you suggested.